Siv3D/src/Siv3D/JSON/SivJSON.cpp

```diff
@@ -524,122 +532,83 @@ namespace s3d
 
 	JSON& JSON::operator =(std::nullptr_t)
 	{
-		if (not m_isValid)
-		{
-			return *this;
-		}
-
 		m_detail->get() = nullptr;
-
+		m_isValid = true;
 		return *this;
 	}
 
-	JSON& JSON::operator =(const JSON& value)
+	JSON& JSON::operator =(const JSON& other)
 	{
-		if (not m_isValid)
-		{
-			return *this;
-		}
-
-		m_detail->get() = value.m_detail->get();
+		m_detail = std::make_shared<detail::JSONDetail>(*other.m_detail);
+		m_isValid = other.m_isValid;
+		return *this;
+	}
 
+	JSON& JSON::operator =(JSON&& other) noexcept
+	{
+		m_detail = std::exchange(other.m_detail, std::make_shared<detail::JSONDetail>());
```

Comment:
```suggestion
		*m_detail = std::exchange(*other.m_detail, detail::JSONDetail{});
```

中身の exchange でいいと思います。